mmv1/third_party/terraform/services/compute/resource_compute_region_security_policy_test.go.tmpl

```diff
@@ -166,4 +166,699 @@ resource "google_compute_region_security_policy" "regionSecPolicy" {
 }
 `, context)
 }
+
+func TestAccComputeRegionSecurityPolicy_regionSecurityPolicyWithRulesBasicUpdate(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckComputeRegionSecurityPolicyDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccComputeRegionSecurityPolicy_withRules(context),
+			},
+			{
+				ResourceName:      "google_compute_region_security_policy.regionSecPolicy",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccComputeRegionSecurityPolicy_withRulesUpdate(context),
```

Comment:
This will ensure that the resource is actually being updated in place (rather than deleted and recreated.

```suggestion
				Config: testAccComputeRegionSecurityPolicy_withRulesUpdate(context),
				ConfigPlanChecks: resource.ConfigPlanChecks{
					PreApply: []plancheck.PlanCheck{
						plancheck.ExpectResourceAction("google_compute_region_security_policy.regionSecPolicy", plancheck.ResourceActionUpdate),
					},
				},
```